src/app/dataset-view/additional-components/dataset-settings-component/tabs/webhooks/create-subscription-modal/create-subscription-modal.component.html

```diff
@@ -0,0 +1,124 @@
+                    >
+                </div>
+            </div>
+
+            <p class="text-muted fs-12">
+                Must be publicly reachable via HTTPS. Localhost and private Ips are not allowed
+            </p>
+        </div>
+        <div class="d-flex flex-column">
+            <label for="" class="form-label">Subscribed events <span class="text-danger">*</span></label>
+            <ng-select
+                [items]="DROPDOWN_LIST"
+                bindLabel="name"
+                bindValue="value"
+                formControlName="eventTypes"
+                [multiple]="true"
+            >
+                <ng-template ng-option-tmp let-item="item" let-index="index">
+                    <label class="fs-14 inline-block"> {{ item.name }} </label>
+                </ng-template>
+            </ng-select>
+            <div *ngIf="eventTypesControl?.invalid && (eventTypesControl?.touched || eventTypesControl?.dirty)">
+                <div *ngIf="eventTypesControl?.errors?.required">
+                    <span class="text-danger fs-12" data-test-id="error-name-pattern">
+                        Event selection is required</span
+                    >
+                </div>
+            </div>
+        </div>
+        <div class="d-flex flex-column mt-4">
+            <label for="" class="form-label">Label </label>
+            <input type="text" data-test-id="target-url" formControlName="label" class="w-100 form-control" />
+            <p class="text-muted fs-12">Optinal unique label to identify this webhook subscription</p>
+        </div>
+    </form>
+    <div class="d-flex flex-column" *ngIf="subscriptionData?.secret">
+        <label for="" class="form-label">Secret </label>
+        <div class="d-flex gap-2">
+            <input
+                type="text"
+                data-test-id="subscription-secret"
+                [title]="secret"
+                [(ngModel)]="secret"
+                class="form-control w-75"
+            />
+            <button type="button" [disabled]="true" class="btn btn-primary" data-test-id="rotate-secret-btn">
+                Rotate secret
+            </button>
+        </div>
+    </div>
+</div>
+
+<div class="px-4 pt-4 pb-2 d-flex" *ngIf="!subscriptionData?.secret">
+    <button type="button" (click)="activeModal.close()" class="btn btn-light flex-fill me-2" data-test-id="save-member">
+        Cancel
+    </button>
+    <button
+        type="button"
+        (click)="createWebhook()"
+        [disabled]="createSubscriptionForm.invalid"
+        class="btn btn-success flex-fill"
+        data-test-id="create-subscription"
+        *ngIf="!subscriptionData"
+    >
+        Create
+    </button>
+    <button
+        type="button"
+        (click)="updateWebhook()"
+        [disabled]="createSubscriptionForm.invalid"
+        class="btn btn-success flex-fill"
+        data-test-id="create-subscription"
+        *ngIf="subscriptionData"
+    >
+        Update
+    </button>
```

Comment:
This markup feels 90% the same. Maybe just customize the button name depending on the condition.